File: 오선민/programmers/P181861.py
'''
문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/120904
시간: 0.10ms
메모리: 10.9MB
PR 제목 : 날짜 사이트_명 문제_번호 사용_언어
'''

def solution(arr):
    x = []
    for i in arr:
        x += [i] * i
    return x

File: 공예진/programmers/P181861.java
//문제: https://school.programmers.co.kr/learn/courses/30/lessons/181861
//시간: 0.12ms
//메모리: 81.2MB


import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int size = 0;
        for (int num : arr) {
            size += num;
        }

        int[] answer = new int[size];
        int idx = 0;
        for (int i = 0; i < arr.length; i++) {
            int num = arr[i];
            Arrays.fill(answer, idx, idx + num, num);
            idx += num;
        }

        return answer;
    }
}